### PR TITLE
[Feature] Support deferred and immediate refresh materialized view

### LINKED ISSUE #16736: Refresh Immediately After Create
## Feature request

To make the materialized view more easier to use, we decide to modify several behaviors:
1. Modify the syntax of create materialized view, add keyword `IMMEDIATE | DEFERRED` to control refresh or not after create
2. The default behavior is changed to `IMMEDIATE`, which would populate the MV after creating
3. Submit an background analyze table task after refreshing the MV
4. Modify the default execution parameter to avoid overload the system: `task_runs_concurrency`, `mv_refresh_mem_limit`

**The create materialized view syntax**
```sql
create materialized view immediate_mv 
refresh deferred async 
distributed by hash(c_1_9) as
select c_1_9, c_1_4 from t1;
```

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #16736

## Problem Summary(Required) ：

Introduce the `IMMEDIATE` and `DEFERRED` refresh for materialized view, like

```sql
create materialized view immediate_mv 
refresh deferred async 
distributed by hash(c_1_9) as
select c_1_9, c_1_4 from t1;
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [x] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 23 / 23 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AsyncRefreshSchemeDesc.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/IncrementalRefreshSchemeDesc.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SyncRefreshSchemeDesc.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ManualRefreshSchemeDesc.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RefreshSchemeDesc.java|3|3|100.00%|[]|

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 54 / 57 (94.74%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/scheduler/SqlTaskRunProcessor.java|1|2|50.00%|[41]|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|5|7|71.43%|[146, 147]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/Alter.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/persist/TaskRunStatus.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AsyncRefreshSchemeDesc.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/SubmitResult.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/IncrementalRefreshSchemeDesc.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SyncRefreshSchemeDesc.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ManualRefreshSchemeDesc.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunContext.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ConnectContext.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/Task.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskBuilder.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/BaseTaskRunProcessor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RefreshSchemeDesc.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRun.java [hunk 1]
```diff
@@ -135,6 +139,15 @@ public boolean executeTaskRun() throws Exception {
             status.setErrorCode(errorCode);
             return false;
         }
+
+        // Execute post task action, but ignore any exception
+        if (StringUtils.isNotEmpty(taskRunContext.getPostRun())) {
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRun.java:144]: What is the performance of the analyze statement?

suggestion:
nit:To put it simply change, better add a switch to decide whether to execute postRun, or only use for MV.
If it is only used for MV, you can use task.getSource to determine the source of MV.

If it is complicated change, I think that this part may also be configurable. For example, if the user understands the details, he can configure whether to analyze full, analyze sample or not analyze after refreshing.

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRun.java (hunk 1 above)
Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRun.java:144]: - Without analyzing the table, the MV rewriter barely works
- It's good to add a switch to configure the analyze behavior, I would do it

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 69 / 77 (89.61%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/scheduler/SqlTaskRunProcessor.java|1|2|50.00%|[41]|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|5|7|71.43%|[147, 148]|
|:large_blue_circle:|com/starrocks/scheduler/TaskBuilder.java|10|14|71.43%|[58, 64, 65, 67]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|5|6|83.33%|[1318]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/Alter.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/persist/TaskRunStatus.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AsyncRefreshSchemeDesc.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/SubmitResult.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/IncrementalRefreshSchemeDesc.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SyncRefreshSchemeDesc.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ManualRefreshSchemeDesc.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunContext.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ConnectContext.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/Task.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/BaseTaskRunProcessor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RefreshSchemeDesc.java|3|3|100.00%|[]|

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 2]
```diff
@@ -833,6 +834,9 @@ public boolean getUseScanBlockCache() {
     @VarAttr(name = ENABLE_RULE_BASED_MATERIALIZED_VIEW_REWRITE)
     private boolean enableRuleBasedMaterializedViewRewrite = true;
 
+    @VarAttr(name = ANALYZE_FOR_MV)
+    private String analyzeForMV = "sample";
+
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:839]: Why not keep the same logic as the olap table?

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:839]: I'm afraid that's too expensive for MV,

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 3]
```diff
@@ -833,6 +834,9 @@ public boolean getUseScanBlockCache() {
     @VarAttr(name = ENABLE_RULE_BASED_MATERIALIZED_VIEW_REWRITE)
     private boolean enableRuleBasedMaterializedViewRewrite = true;
 
+    @VarAttr(name = ANALYZE_FOR_MV)
+    private String analyzeForMV = "sample";
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:838]:
```suggestion
    private String analyzeTypeForMV = "sample";
```

Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:838]: maybe we should provide the users valid options? sample/full?

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:838]: done

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Review by dirtysalt — APPROVED** [review]

**Review by Dshadowzh — APPROVED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**